t/t0005-telemetry.t

```diff
@@ -0,0 +1,162 @@
+#!/bin/sh
+
+test_description='Test Powerman telemetry output'
+
+. `dirname $0`/sharness.sh
+
+powermand=$SHARNESS_BUILD_DIRECTORY/src/powerman/powermand
+powerman=$SHARNESS_BUILD_DIRECTORY/src/powerman/powerman
+vpcd=$SHARNESS_BUILD_DIRECTORY/test/vpcd
+vpcdev=$SHARNESS_TEST_SRCDIR/../etc/devices/vpc.dev
+
+# Use port = 11000 + test number
+# That way there won't be port conflicts with make -j
+testaddr=localhost:11005
+
+test_expect_success 'create test powerman.conf' '
+	cat >powerman.conf <<-EOT
+	include "$vpcdev"
+	listen "$testaddr"
+	device "test0" "vpc" "$vpcd |&"
+	node "t[0-15]" "test0"
+	EOT
+'
+test_expect_success 'start powerman daemon and wait for it to start' '
+	$powermand -c powerman.conf -f &
+	echo $! >powermand.pid &&
+	$powerman --retry-connect=100 --server-host=$testaddr -q >/dev/null
+'
+test_expect_success 'powerman -q -T works' '
+	$powerman -h $testaddr -q -T >tel.out
+'
+test_expect_success 'telemetry output contains send' '
+	grep -q "send(test0):" tel.out
+'
+test_expect_success 'telemetry output contains recv' '
+	grep -q "recv(test0):" tel.out
+'
+test_expect_success 'telemetry has expected size' '
+	test $(stat -c%s tel.out) -eq 576
+'
+test_expect_success 'stop powerman daemon' '
+	kill -15 $(cat powermand.pid) &&
+	wait
+'
+
+generate_config_spew() {
+	local spew=$1
+	cat <<-EOT
+	specification "vpc" {
+	    timeout 	5.0
+	        plug name { "0" "1" "2" "3" "4" "5" "6" "7" "8"
+	                    "9" "10" "11" "12" "13" "14" "15" }
+	    script login {
+	        send "login\n"
+	        expect "[0-9]* OK\n"
+	        expect "[0-9]* vpc> "
+	    }
+	    script logout {
+	        send "logoff\n"
+	        expect "[0-9]* OK\n"
+	    }
+	    script status_all {
+	        send "spew $spew\n"  # noise generator
+	        expect "[0-9]* OK\n"
+	        expect "[0-9]* vpc> "
+	        send "stat *\n"
+	        foreachplug {
+	            expect "plug ([0-9]+): (ON|OFF)\n"
+	            setplugstate \$1 \$2 on="ON" off="OFF"
+	        }
+	        expect "[0-9]* OK\n"
+	        expect "[0-9]* vpc> "
+	    }
+        }
+	listen "$testaddr"
+	device "test0" "vpc" "$vpcd |&"
+	node "t[0-15]" "test0"
+	EOT
+}
+test_expect_success 'create test powerman.conf with extra telemetry output < 1K' '
+	generate_config_spew 16 >powerman2.conf
+'
+test_expect_success 'start powerman daemon and wait for it to start' '
+	$powermand -c powerman2.conf -f &
+	echo $! >powermand2.pid &&
+	$powerman --retry-connect=100 --server-host=$testaddr -q >/dev/null
+'
+test_expect_success 'powerman -q -T works' '
+	$powerman -h $testaddr -q -T >tel2.out
+'
+test_expect_success 'telemetry has expected size' '
+	test $(stat -c%s tel2.out) -eq 1926
+'
```

Comment:
i think this approach is fine, but maybe a comment that says generally what is expected of the output?  Since to the uniniated, they got no idea what this test does and what "spew" does.  Maybe something like

```
send> on
abcdefghfdfdfadsfdsafdsafdsfsda .... (X times)
recv> ...
on: ...
off: ...
```